antarest/study/storage/variantstudy/variant_study_service.py

```diff
@@ -852,34 +855,36 @@ def _generate(
         )
         results = self._generate_snapshot(
             variant_study=variant_study,
-            dest_path=dest_path,
+            dst_path=dst_path,
             notifier=notifier,
             from_command_index=command_start_index,
         )
 
         if unmanaged_user_config:
-            logger.info("Restoring previous unamanaged user config")
-            if dest_path.exists():
-                if (dest_path / "user").exists():
+            logger.info("Restoring previous unmanaged user config")
+            if dst_path.exists():
+                if (dst_path / "user").exists():
                     logger.warning(
-                        "Existing unamanaged user config. It will be overwritten."
+                        "Existing unmanaged user config. It will be overwritten."
                     )
-                    shutil.rmtree((dest_path / "user"))
-                shutil.copytree(unmanaged_user_config, dest_path / "user")
+                    shutil.rmtree((dst_path / "user"))
+                shutil.copytree(unmanaged_user_config, dst_path / "user")
             else:
                 logger.warning("Destination snapshot doesn't exist !")
             shutil.rmtree(unmanaged_user_config, ignore_errors=True)
```

Comment:
```python
if not unmanaged_config:
    return

logger.info("Restoring previous unmanaged user configuration")

if not dest_path.exists():
    logger.warning("Destination snapshot doesn't exist!")
    return

user_config_path = dest_path / "user"
if user_config_path.exists():
    logger.warning("Existing unmanaged user configuration will be overwritten.")
    shutil.rmtree(user_config_path)

shutil.copytree(unmanaged_config, user_config_path)
shutil.rmtree(unmanaged_config, ignore_errors=True)
```
here I've used early returns to reduce the level of nesting, also created a `user_path` variable to avoid repetition. + `dst_path` > `dest_path`

Reply:
Intéressant. Cela dit, je n'ai fait que renommer les variables. Je garde ça pour une prochaine fois.